security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,207 @@
+  - [FM8: Smart contracts relying on sender check no longer ensures sender does not have code](#fm8-smart-contracts-relying-on-sender-check-no-longer-ensures-sender-does-not-have-code)
+  - [Generic items we need to take into account:](#generic-items-we-need-to-take-into-account)
+- [Action Items](#action-items)
+- [Audit Requirements](#audit-requirements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+<!-- _Italics are used to indicate things that need to be replaced._ -->
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | Julian Meyer                                       |
+| Created at         | 2025-02-10                                         |
+| Initial Reviewers  | refcell                                            |
+| Need Approval From |                                                    |
+| Status             | In Review 🔎                                       |
+<!-- 
+> [!NOTE]
+> 📢 Remember:
+>
+> - The single approver in the “Need Approval From” must be from the Security team.
+> - Maintain the “Status” property accordingly. An FMA document can have the following statuses:
+>   - **Draft 📝:** Doc is created but not yet ready for review.
+>   - **In Review 🔎:** Security is reviewing, and Engineering is iterating on the design. A checklist of action items will be created during this phase.
+>   - **Implementing Actions 🛫:** Security has signed off on the content of the document, including the resulting action items. Engineering is responsible for implementing the action items, and updating the checklist.
+>   - **Final 👍:** Security will transition the status of the document to Final once all action items are completed.
+
+> [!TIP]
+> Guidelines for writing a good analysis, and what the reviewer will look for:
+>
+> - Show your work: Include steps and tools for each conclusion.
+> - Completeness of risks considered.
+> - Include both implementation and operational failure modes
+> - Provide references to support the reviewer.
+> - The size of the document will likely be proportional to the project's complexity.
+> - The ultimate goal of this document is to identify action items to improve the security of the project. The FMA review process can be accelerated by proactively identifying action items during the writing process. -->
+
+## Introduction
+
+This document covers implementation of Pectra features in the OP stack. This requires a hard fork due to new transaction types like EIP-7702 transactions. This does not cover generic hardfork related security concerns, which will be addressed in a separate document.
+
+Below is a list of EIPs that are enabled with Pectra, and a quick description of spec changes needed to support the EIP.
+
+- EIP-7702: Set code transaction
```

Comment:
For here, I would like to know if this is possible to make a table like this: 

https://github.com/ethereum-optimism/design-docs/blob/879b1c4d725bf0769c4a589faff7e9a4d468ff63/security/fma-isthmus-l2-pectra.md

And also add one column that is "Is that require custom code"

The idea is to identify if this required custom code to support this EIP or we just merge from UPstream-Geth

Reply:
Added this here: https://github.com/ethereum-optimism/design-docs/pull/201/commits/5401b30af365b9ca4f9fcad0569dff2d3bf31616

@geoknee or @sebastianst - would be awesome if you could review the components changed for each of these! Not confident I know exactly what op-geth changes were required for each of these.

Reply:
Thank you, I am pinging them to see how we can do that. 
I don't know if https://op-geth.optimism.io/ for now would be enough accurate. 
I will update the content when I got the answer probably tmw.

Reply:
Yes https://op-geth.optimism.io/ shows the up-to-date diff of op-geth to upstream geth.

Reply:
This seems to be too much work, and not easy to use with op-geth.optimism.io since this is showing all the diff against geth but we don't have an obvious way to get isthmus changes only...